server/src/controllers/clients.controller.js

```diff
@@ -0,0 +1,120 @@
+const CrudRestController = require('./crud-rest.controller');
+const ClientModel = require('../models/client.model');
+const ValidationData = require('../helper/validationIncomingData');
+
+class ClientsController extends CrudRestController {
+
+  registerRoutes() {
+    this.router.get("/byName", this.getByName.bind(this));
+    super.registerRoutes();
+  }
+
+  /**
+   * List resources
+   */
+  list(req, res) {
+    ClientModel.find({ active: true }, (error, client) => {
+
+      if (error) {
+        this._error(res, "Failed to get all clients.", 500);
+        return;
+      }
+      var data = client ? client : {};
+      this._success(res, data);
+    })
+  }
+
+  /**
+   * Create resource
+   */
+  create(req, res) {
+
+    var validation = ClientModel.validateCreate(req.body);
+    if (validation.error) {
+      this._error(res, validation.error.details, 422);
+      return;
+    }
+
+    // check if client exist
+    ClientModel.find({ $or: [{ name: req.body.name }] }, (err, client) => {
+      if (client.length > 0) {
+        this._error(res, "Client already exist!")
+        return
+      }
+
+      var newClient = new ClientModel({
+        name: req.body.name,
+        employees: req.body.employees,
+        active: true
+      });
+
+      newClient.save(function (error, fluffy) {
```

Comment:
this should be an arrow function, like 
`(error, fluffy) => { `
to prevent error with 'this' on lines 53 and 56

Reply:
fixed